interface/patient_file/summary/demographics_full.php

```diff
@@ -792,7 +806,56 @@ function end_group() {
     $("select").change(function() {
         checkSkipConditions();
     });
+
+//This code deals with demographics before save action -
+<?php if($hook):?>
+
+	var f = $("form");
+
+	// Use hook to open the controller and get the new patient validation .
+	// when no params are sent this window will be closed from the zend controller.
+	var url ='<?php echo  $GLOBALS['web_root']."/interface/modules/zend_modules/public/patientvalidation?closeBeforeOpening=1";?>';
+	$("#submit").attr("type","button");
+	$("#submit").attr("name","btnSubmit");
+	$("#submit").attr("id","btnSubmit");
+	$("#btnSubmit").click(function( event ) {
+
+
+		<?php
+		// D in edit_options indicates the field is used in duplication checking.
+		// This constructs a list of the names of those fields.
+		$mflist = "";
+		$mfres = sqlStatement("SELECT field_id FROM layout_options " .
+			"WHERE form_id = 'DEM' AND uor > 0 AND field_id != '' AND " .
+			"edit_options LIKE '%D%' " .
+			"ORDER BY group_name, seq");
+		while ($mfrow = sqlFetchArray($mfres)) {
+			$field_id  = $mfrow['field_id'];
+			if (strpos($field_id, 'em_') === 0) continue;
+			if (!empty($mflist)) $mflist .= ",";
+			$mflist .= "'" . htmlentities($field_id) . "'";
```

Comment:
don't use htmlentities(this is encoding specific)
instead for html escaping in this case, use the attr() function